Modules/CPV/src/PedestalTask.cxx

```diff
@@ -0,0 +1,483 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   PedestalTask.cxx
+/// \author Sergey Evdokimov
+///
+
+#include <TCanvas.h>
+#include <TH1.h>
+#include <TH2.h>
+#include <TF1.h>
+#include <TSpectrum.h>
+
+#include "QualityControl/QcInfoLogger.h"
+#include "CPV/PedestalTask.h"
+#include <Framework/InputRecord.h>
+#include "DataFormatsCPV/TriggerRecord.h"
+
+namespace o2::quality_control_modules::cpv
+{
+
+PedestalTask::~PedestalTask()
+{
+  for (int i = 0; i < kNHist1D; i++) {
+    if (mHist1D[i]) {
+      mHist1D[i]->Delete();
```

Comment:
is there a particular reason not to use `delete mHist1D[i]` ?

Reply:
no, there is no reason.

Reply:
I tried to understand the difference between histo->Delete() and delete histo and they both call destructor but first case contains additional check IsOnHeap(). Whatever this check is I prefer to get rid of it just to be 100% sure histo is deleted. In future commits I will replace `mHist1D[i]->Delete()` by `delete mHist1D[i]` as you suggested.